chef/cookbooks/crowbar/recipes/crowbar-upgrade.rb

```diff
@@ -27,6 +27,41 @@
 upgrade_step = node["crowbar_wall"]["crowbar_upgrade_step"] || "none"
 
 case upgrade_step
+
+when "crowbar_upgrade"
+
+  # Disable openstack services
+  # We don't know which openstack services are enabled on the node and
+  # collecting that information via the attributes provided by chef is
+  # rather complicated. So instead we fall back to a simple bash hack
+
+  bash "disable_openstack_services" do
+    code <<-EOF
+      for i in /etc/init.d/openstack-* /etc/init.d/openvswitch-switch /etc/init.d/ovs-usurp-config-* /etc/init.d/drbd /etc/init.d/openais;
+      do
+        if test -e $i
+        then
+          initscript=`basename $i`
+          chkconfig -d $initscript
+        fi
+      done
+    EOF
+    only_if { node[:platform] == "suse" }
+  end
+
+  # Disable crowbar-join
+  service "crowbar_join" do
+    # do not stop it, it would change node's state
+    action :disable
+    only_if { node[:platform] == "suse" }
```

Comment:
we were wondering if this check is needed as we are already checking the platform_family at the beginning of the recipe.

basically I think it is correct in case a service could have different names in different suse products.
any other opinions?